src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -708,5 +744,156 @@ public static nuint RotateRight(nuint value, int offset)
             return (nuint)RotateRight((uint)value, offset);
 #endif
         }
+
+        /// <summary>
+        /// Calculates the CRC (Cyclic redundancy check) checksum.
+        ///
+        /// Uses following Hardware Intrinsics, if supported:
+        /// * uint32_t __crc32b (uint32_t a, uint8_t b)
+        /// * uint32_t __crc32cb (uint32_t a, uint8_t b)
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The checksum data</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, byte data)
+        {
+            if (Sse42.IsSupported)
+            {
+                // uint32_t __crc32b (uint32_t a, uint8_t b)
+                return Sse42.Crc32(crc, data);
+            }
+            if (Crc32.IsSupported)
+            {
+                // uint32_t __crc32cb (uint32_t a, uint8_t b)
+                return Crc32.ComputeCrc32C(crc, data);
+            }
+
+
+            // Software fallback
+            int tableIndex = (int)((crc ^ data) & 0xFF);
+            crc = CrcTable[tableIndex] ^ (crc >> 8);
+
+            return crc ^ 0xFFFFFFFF;
+        }
+
+        /// <summary>
+        /// Calculates the CRC (Cyclic redundancy check) checksum.
+        ///
+        /// Uses following Hardware Intrinsics, if supported:
+        /// * unsigned int _mm_crc32_u16 (unsigned int crc, unsigned short v)
+        /// * uint32_t __crc32ch (uint32_t a, uint16_t b)
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The checksum data</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, ushort data)
+        {
+            if (Sse42.IsSupported)
+            {
+                // unsigned int _mm_crc32_u16 (unsigned int crc, unsigned short v)
+                return Sse42.Crc32(crc, data);
+            }
+            if (Crc32.IsSupported)
+            {
+                // uint32_t __crc32ch (uint32_t a, uint16_t b)
+                return Crc32.ComputeCrc32C(crc, data);
+            }
+
+
+            // Software fallback
+
+            Span<byte> bytes = stackalloc byte[sizeof(ushort)];
+            Unsafe.As<byte, ushort>(ref bytes[0]) = data;
```

Comment:
```suggestion
            Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(bytes), data);
```

Reply:
What benefit does this provide? Does this generate different JIT assembly code?

I prefer in readability the `Unsafe.As` variant and `BitConverter.GetBytes` also uses the `Unsafe.As` approach.

Reply:
The generated JIT Asm is the same. See [SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgAYBYAKDIEYbjyACYugOgCUBXAOwwCW+GOwCS/GFAgAHAMqSAbgLAxcAbkYtxGSb2wAbLn0HD2AYQj5pA/ZPlQlK9TUYpmAMQFRcGABTdcAAtoDGYAE2wMbABKZgBvGmYk5llpbF4AHmAATx0APmYcnVxmAF5mH2wwAGsDfQgwQtyYAG1cAQAvGAgAM38gkOiAXQ1qZOYAVV5cbB6RAEFcLOa0ZgDgqAw831gepuKW8iHY8oio0YBfF1o3eUheMP6N0LOY+MTk1PTl/P3VMoqURqdQafzanW6fXWgxGHySUxmc3YAHUoAIdFMDAIAOa8GCPXbMACyMHw0GyROw3kCBnYAHEYBhODA5rBeCpfEVVNFVq9opcgA=),

Reply:
In current implementations of JIT it's the same generated code. With `WriteUnaligned` it's explicit that an unaligned write needs to be done, so it's better code-hygiene and definitely future proof iif the JIT would ever emit different machine code for `Unsafe.As`.